shared/rocroller/client/src/gemm.cpp

```diff
@@ -62,6 +62,47 @@ enum ReturnCodes : int
     SolutionNotSupportedOnArch = 3
 };
 
+template <typename T>
+class RotatingBuffer
+{
+public:
+    RotatingBuffer(size_t numElems, size_t rotationSize, const std::vector<T>& hostData = {})
```

Comment:
```suggestion
    RotatingBuffer(const std::vector<T>& hostData, size_t rotationSize)
```

Two points here:
(1) I don't think we need to support a default `std::vector<T>&`; we should always pass one in.
(2) if we have a reference to `hostData` now, we shouldn't pass the size separately.

Reply:
I agree! I have updated the implementation accordingly in the commit 4a9aa6c :
- The RotatingBuffer constructor now always takes a host data vector, and derives its size internally. This removes the need for a separate numElems argument and avoids supporting an unused default case.